backend/src/api/routes/project.ts

```diff
@@ -0,0 +1,218 @@
+import type { Request, Response } from 'express';
+import type { ProjectMetadata } from '../../types/ProjectMetadata';
+import { firebaseAdmin } from '../../admin';
+import type { UserToken } from '../../types/UserToken';
+import type { ProjectId } from '../../types/ProjectId';
+import type { User } from '../../types/User';
+import { verifyUidToken } from '../../util/verifyUidToken';
+import type { Uid } from '../../types/Uid';
+import { verifyProjectAccess } from '../../util/verifyProjectAccess';
+
+interface GetProjectParams {
+  uid: string;
+  userToken: UserToken;
+  projectIds: Array<ProjectId>;
+}
+
+interface ProjectLocale {
+  code: string;
+  name: string;
+}
+
+interface ProjectMetadataResponse {
+  id: ProjectId;
+  locales?: Record<string, ProjectLocale> | Array<ProjectLocale>;
+  name: string;
+  relativeBasePath: string;
+  userRoles: Record<string, 'owner' | 'editor'>;
+  /**
+   * @deprecated
+   * */
+  users?: Record<string, Uid>;
+}
+
+export async function getProjects(request: Request, response: Response): Promise<void> {
+  const { uid, userToken, projectIds } = request.query as Partial<GetProjectParams>;
+
+  if (!uid || !userToken) {
+    response.status(400).send({
+      error: {
+        code: 'error.missing_params',
+        message: 'Not all params are present',
+      },
+    });
+    return;
+  }
+
+  const projectsMetadata = await firebaseAdmin
+    .database()
+    .ref('/projectMetadata')
+    .get()
+    .then((data) => data.toJSON() as Record<string, ProjectMetadataResponse>);
+
+  let projects = [...Object.values(projectsMetadata)];
+
+  projects = projects.filter((project) => {
```

Comment:
the `let` can be a `const` if you do it like this:

```suggestion
 const projects = Object.values(projectsMetadata).filter((project) => {
```